src/almanack/metrics/data.py

```diff
@@ -770,3 +783,141 @@ def get_github_build_metrics(
+
+                    # Retrieve the timestamp
+                    timestamp = datetime.fromisoformat(
+                        coverage_data["meta"]["timestamp"]
+                    )
+
+                elif coverage_file.endswith(".xml"):
+                    # Parse XML coverage data (using defusedxml for safety)
+                    tree = ET.parse(file_path)
+                    root = tree.getroot()
+
+                    # Extract the total lines and executed lines directly from the root element
+                    total_lines = int(root.attrib.get("lines-valid", 0))
+                    executed_lines = int(root.attrib.get("lines-covered", 0))
+
+                    # Optional: parse timestamp if present in XML attributes
+                    timestamp_str = root.attrib.get("timestamp")
+                    timestamp = datetime.fromtimestamp(
+                        float(timestamp_str)
+                        / 1000  # Convert from milliseconds to seconds
+                    )
+
+                elif coverage_file.endswith(".lcov"):
+                    # Parse LCOV coverage data
+                    with open(file_path, "r") as f:
+                        for line in f:
+                            working_line = line.strip()
+                            if working_line.startswith(
+                                "DA:"
+                            ):  # DA:<line number>,<execution count>
+                                _, execution_count = working_line.split(":")[1].split(
+                                    ","
+                                )
+                                total_lines += 1
+                                if int(execution_count) > 0:
+                                    executed_lines += 1
+
+                    # Determine the latest commit date for the LCOV file
+                    timestamp = next(
+                        (
+                            datetime.fromtimestamp(commit.commit_time)
+                            for commit in repo.walk(
+                                repo.head.target, pygit2.GIT_SORT_TIME
+                            )
+                            if coverage_object.name in commit.tree
+                        ),
+                        None,
+                    )
+
+                # Calculate coverage percentage
+                coverage_percentage = (
+                    (executed_lines / total_lines * 100) if total_lines > 0 else 0.0
+                )
+
+                return {
+                    "code_coverage_percent": coverage_percentage,
+                    "date_of_last_coverage_run": timestamp.strftime(
+                        "%Y-%m-%dT%H:%M:%S.%fZ"
+                    ),
+                    "total_lines": total_lines,
+                    "executed_lines": executed_lines,
+                }
+
+            except Exception as e:
+                LOGGER.warning(f"Error reading {coverage_file}: {e}")
```

Comment:
Is it impossibly complex to be more specific here? (would it even be helpful?)

Are there scenarios in which a repo will have an old version of these coverage files? I imagine that we wouldn't modify our processing if, for example, an index integer was different in version 2 vs 3, we would just throw this error?

Reply:
> Is it impossibly complex to be more specific here? (would it even be helpful?)

Thanks great comment! There's a lot of complexity within this function that could result in many different types of errors or exceptions (for example, XML and JSON parsing have their own unique exception cases).

> Are there scenarios in which a repo will have an old version of these coverage files?

Yes, for sure, good thoughts! I tried to make this more relevant in the data using the `date_of_last_coverage_run` key value pair. We could take this further by adding a days of difference between the latest comment and that key value - I'll add that in as it seems like this could be very important.

Reply:
Added in a new metric to compare the days of difference between the latest coverage run and the latest commit as `repo-days-between-last-coverage-run-latest-commit`.